### PR TITLE
Fixed honking angels

### PR DESCRIPTION
**Pull request by FlavoredCactus**
-Fixed crashs relating to sucking power from space
-adds a neck snaping sound effect.
-extends the character setup menu a few pixels to accomodate the new
options

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -999,6 +999,27 @@ About the new airlock wires panel:
 		..()
 	return
 
+	//honking angel claw doors open
```
Comment (xXTheDracheXx) [inline code/game/machinery/doors/airlock.dm]: I believe this should really be in the angel code not in the door code.

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -49,6 +49,20 @@
 			return
 	return
 
+//honking angel stuff
```
Comment (xXTheDracheXx) [inline code/game/machinery/doors/firedoor.dm]: Same as above.

File: code/modules/mob/honking_angel.dm [hunk 3]
```diff
@@ -347,6 +347,16 @@
 	set name = "Power Disruption"
 	set desc = "Disrupt and drain power from the local area."
 	set category = "H.Angel"
+	//find current area
+	var/area/A = src.loc
```
Comment (xXTheDracheXx) [inline code/modules/mob/honking_angel.dm]: Use get_area, not using loc which isn't even a guaruantee it is an area.

File: code/modules/mob/honking_angel.dm [hunk 4]
```diff
@@ -527,7 +533,7 @@
...
+	src << "\black Honking Angels are an ancient species of statuesque humanoids. Not to be confused with their weeping cousins, though they do share some similarities. Angels are predatory by nature and feed off forms of energy be it radiation, electricity, or even stealing energy from living beings. It's worth it to note that angels are sadistic and very intellegent. They find great pleasure harrassing and causing their prey anguish before striking. Angels have evolved with a very special defensive ability. They will involuntarily quantum lock when they are under observation to protect themselves to ALMOST all forms of harm. They also possess a voracious appetite and can starve to death in the span of a single round. In fact if you are reading this you are actually on your way to starvation right now so there is no time to waste. "
```
Comment (xXTheDracheXx) [inline code/modules/mob/honking_angel.dm]: This \black shouldn't be necessary as it's black by default.

**Comment by xXTheDracheXx** [comment]
Major complaints lie in that you put code that should really be in one file only, into many many files. This is pretty bad form imo and just clutters up other atom's code with snowflake stuff.

**Comment by LLADonZombie** [comment]
You are right about the two door parts. Flavo was just porting it for me the way I originally wrote it.
You can move the two door parts into the main file flavo. Also it really is a porting because whoever "ported it" originally left behind the parts of the code that couldn't be in the main file like the changes to the simple animal attack procs.

**Comment by xXTheDracheXx** [comment]
Doesn't really matter. Anyway, I'm not merging this and I advise against merging this until it's refactored.

**Comment by LLADonZombie** [comment]
Re-factoring complete

File: code/modules/mob/honking_angel.dm [hunk 5]
```diff
@@ -347,6 +347,16 @@
...
 	set desc = "Disrupt and drain power from the local area."
 	set category = "H.Angel"
+	//find current area
+	var/area/A = get_area(src.loc)
+	//A = A.loc
+	if (!( istype(A, /area))) // If its not a area, stop!
+		return
+	if(!A.requires_power) // If it doesn't require power, stop!
+		return
+	if (istype(A, /area/space)) // If it is space, LOVE OF GOD PLEASE STOP!
```
Comment (xXTheDracheXx) [inline code/modules/mob/honking_angel.dm]: I understand it's probably not you who coded it, but hardcoding is bad. There should be a variable that says that there isn't any power in the area, or that it doesn't need power. Should check for those. (Unless it's an issue, you don't need to change it.)

**Pull request merged**